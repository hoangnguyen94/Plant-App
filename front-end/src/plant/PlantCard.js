//PlantCard Component
import React from "react";
import Toxicity from "../toxicity/Toxicity";
import ExpandMoreIcon from "@mui/icons-material/ExpandMore";

import {
  Accordion,
  AccordionDetails,
  AccordionSummary,
  Typography,
  Box,
  Container,
  CircularProgress,
  Paper,
  StepConnector
} from "@mui/material";

import "./PlantCard.css";


function PlantCard({ selectedPlantData }) {

  console.log(JSON.stringify(selectedPlantData))
  if ( selectedPlantData )
  {

    return (
      <Container sx={{ width: { md: 800 } }}>
        <Box mt={2} />
        <Container sx={{ textAlign: "center" }}>
          <div className="plant-img-container">
            <img src={selectedPlantData.image_url} alt={"plant_image"} />
          </div>

        </Container>

        <Typography paddingTop={4} variant="h4" margin={2} sx={{ textAlign: "center" }}>
          {selectedPlantData.name}
        </Typography>
        <Typography
          variant="subtitle2"
          margin={2}
          sx={{ textAlign: "center", color: "text.secondary" }}
        >
          <i>
            {selectedPlantData.sci_name}
          </i>
        </Typography>

        <Accordion
          elevation={3}>
          <AccordionSummary
            expandIcon={<ExpandMoreIcon />}
            aria-controls="panel1bh-content"
            id="panel1bh-header"
          >
            <Typography sx={{ width: "33%", flexShrink: 0 }}>
              Details
            </Typography>
            <Typography sx={{ color: "text.secondary" }}>
              Learn more about this plant
            </Typography>
          </AccordionSummary>
          <AccordionDetails>

            <Typography variant="body1" sx={{ color: "text.secondary" }}>
              Family:
            </Typography>
            {selectedPlantData.family} <br /><br />

            <Typography variant="body1" sx={{ color: "text.secondary" }}>
              Common names:

            </Typography>

            {selectedPlantData.common_names}<br /><br />
            <StepConnector />
            <br />
            <Typography sx={{ color: "text.secondary" }}>
              Additional Information:
              <br />
            </Typography>
            <a href={selectedPlantData.aspca_url} target="_blank" rel="noopener noreferrer">
              {selectedPlantData.aspca_url}
            </a>
          </AccordionDetails>
        </Accordion>

        <Box mb={5} />
        <Paper elevation={4}>
          <Toxicity selectedPlantData={selectedPlantData} />
        </Paper>

        <Box mb={15} />


      </Container>
    );

  }
  else
  {
    return (
      <Box sx={{ display: 'flex' }}>
        <CircularProgress />
      </Box>
    )
  }
}

export default PlantCard;
