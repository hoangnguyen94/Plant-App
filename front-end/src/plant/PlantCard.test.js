import React from "react";
import { render } from "@testing-library/react";
import PlantCard from "./PlantCard";

test("renders PlantCard component correctly", () => {
	const selectedPlantData = {
	  image_url: "path/to/image",
	  name: "Plant Name",
	  sci_name: "Scientific Name",
	  family: "Plant Family",
	  common_names: ["Common Name 1", "Common Name 2"],
	  aspca_url: "https://example.com",
	  animals: [
		{ name: "Animal 1", toxic: true },
		{ name: "Animal 2", toxic: false },
	  ],
	};
  
	const { container } = render(<PlantCard selectedPlantData={selectedPlantData} />);
  
	expect(container).toMatchSnapshot();
  });
  