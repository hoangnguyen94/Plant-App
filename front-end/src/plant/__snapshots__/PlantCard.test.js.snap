// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PlantCard component correctly 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-19acubu-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-1yuhvjn"
    />
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-2b8hh-MuiContainer-root"
    >
      <div
        class="plant-img-container"
      >
        <img
          alt="plant_image"
          src="path/to/image"
        />
      </div>
    </div>
    <h4
      class="MuiTypography-root MuiTypography-h4 css-9cq3s9-MuiTypography-root"
    >
      Plant Name
    </h4>
    <h6
      class="MuiTypography-root MuiTypography-subtitle2 css-10uhfgf-MuiTypography-root"
    >
      <i>
        Scientific Name
      </i>
    </h6>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-mb1vyy-MuiPaper-root-MuiAccordion-root"
    >
      <div
        aria-controls="panel1bh-content"
        aria-expanded="false"
        class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-sh22l5-MuiButtonBase-root-MuiAccordionSummary-root"
        id="panel1bh-header"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1m4ljmh-MuiTypography-root"
          >
            Details
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
          >
            Learn more about this plant
          </p>
        </div>
        <div
          class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ExpandMoreIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </div>
      </div>
      <div
        class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
          >
            <div
              aria-labelledby="panel1bh-header"
              class="MuiAccordion-region"
              id="panel1bh-content"
              role="region"
            >
              <div
                class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
                >
                  Family:
                </p>
                Plant Family
                 
                <br />
                <br />
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
                >
                  Common names:
                </p>
                Common Name 1
                Common Name 2
                <br />
                <br />
                <div
                  class="MuiStepConnector-root MuiStepConnector-horizontal css-j5w0w9-MuiStepConnector-root"
                >
                  <span
                    class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
                  />
                </div>
                <br />
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
                >
                  Additional Information:
                  <br />
                </p>
                <a
                  href="https://example.com"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  https://example.com
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-10ib5jr"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation4 css-11mde6h-MuiPaper-root"
    >
      <div
        align="center"
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-fakinf-MuiTypography-root"
        >
          Animal Toxicity
        </h4>
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 css-10uhfgf-MuiTypography-root"
        >
          <i>
            It may be dangerous to the animals displayed below if consumed: 
          </i>
        </h6>
        <div
          align="center"
          class="MuiBox-root css-n9x58s"
        >
          <span
            class="MuiTypography-root MuiTypography-h1 css-sqoje6-MuiTypography-root"
          />
        </div>
        <div
          class="MuiBox-root css-10ib5jr"
        />
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-gdthc8-MuiTypography-root"
          >
            Clinical signs:
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-51tpa5-MuiTypography-root"
          />
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-1y3h0mg"
    />
  </div>
</div>
`;
